### PR TITLE
chore(demo): `DocAPIItem` value reflect in URL

### LINKED ISSUE #9206: 🐞 - `Block`, `Button` in `Demo` change `API` properties not reflected in URL
### Code Example URL

https://taiga-ui.dev/components/button/API

### Description

Related with #9057 

https://github.com/user-attachments/assets/691bf6ab-eb2f-46f6-86cf-40ae2152ade6

### Taiga UI version

4.8.1

### Browser Used

Chrome

### OS Used

macOS

### PR DESCRIPTION
**Pull request by mdlufy**
Fixes #9206

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/demo-playwright/tests/core/button/button.spec.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ test.describe('Button', () => {
     test('darkMode=true + appearance=icon + hovered state', async ({page}) => {
         await tuiGoto(
             page,
-            '/components/button/API?darkMode=true&appearance=icon&icon=tuiIconEyeOff',
+            '/components/button/API?darkMode=true&appearance=icon&iconStart=@tui.eye-off',
```
[the remark this answers is not in the document]
Comment (waterplea) [reply projects/demo-playwright/tests/core/button/button.spec.ts:8]: Let's add it to the demo, it's basic grey text, I think we can show it there. It is sometimes used for pseudo links.

File: projects/addon-doc/components/api/api-item.component.ts [hunk 2]
```diff
@@ -56,4 +69,59 @@ export class TuiDocAPIItem<T> {
...
+        this.parseParams(this.activatedRoute.snapshot.queryParams);
+    }
+
+    public onValueChange(value: T): void {
+        this.value = value;
+        this.valueChange.emit(value);
+        this.setQueryParam(value);
+    }
+
+    @tuiPure
```
Comment (waterplea) [inline projects/addon-doc/components/api/api-item.component.ts:83]: No need for memoization, it is not used in a getter

**Review by waterplea — APPROVED** [review]

File: projects/demo-playwright/tests/core/button/button.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (mdlufy) [reply projects/demo-playwright/tests/core/button/button.spec.ts:8]: Ok, i'll add it in separate PR

File: projects/addon-doc/components/api/api-item.component.ts (hunk 2 above)
Comment (mdlufy) [reply projects/addon-doc/components/api/api-item.component.ts]: removed

**Pull request merged**